packages/grafana-ui/src/utils/colors.ts

```diff
@@ -112,8 +112,15 @@ function hslToHex(color: tinycolor.ColorFormats.HSLA) {
   return tinycolor(color).toHexString();
 }
 
-export function getTextColorForBackground(color: string) {
-  const b = tinycolor(color).getBrightness();
+export function getTextColorForBackground(color: string, themeIsDark: boolean) {
+  const tcolor = tinycolor(color);
+  const b = tcolor.getBrightness();
+  const a = tcolor.getAlpha();
+
+  if (a < 0.3) {
+    return themeIsDark ? 'rgb(247, 248, 250)' : 'rgb(32, 34, 38)';
```

Comment:
Isn't the `b > 180` basically the same thing as `themeIsDark`? I'm asking because this is a breaking change of the `getTextColorForBackground` function since the `themeIsDark` is required.

Reply:
As far as I understandd, the `b` component is strictly related to the background color hex code and does not take into account what the theme settings are. I understand that this is a breaking change. In this case I guess the cleanest solution is to create a new function that takes the transparency component into account and use that internally.